Add h265 encoding

DarthBubi opened the pull request:
Add h265 stream from OAK-D as a replacement to raw images.

DarthBubi committed:
Add h265_node as a first draft

DarthBubi committed:
Minimal working node for h265 decoded streams

DarthBubi committed:
Add hardware device support for h265_node

DarthBubi committed:
Test hw-accelerated decoding

DarthBubi committed:
Add hardware device type configuration to h265_node launch file

DarthBubi committed:
Current dev adds debugging and semi-stable h264

DarthBubi committed:
Add video codec launch argument to h265_node

DarthBubi committed:
Improve H.265 decoding handling

DarthBubi committed:
Add mjpeg node for testing

DarthBubi committed:
Add first version of accelerated gstreamer nvjpeg decoding
This crashes if gstreamer is available and enabled

Cirromulus committed:
Slightly re-arranged logic that I could not test

Cirromulus committed:
We found out that nvidia is lazy and nvjpegdec can't be used when CV functions use the normal libjpeg

Commit:
Add more verbosity and parameters
hw acclerated decoding still fails

Commit:
Finally fix gstreamer pipeline for decoding mjpeg

Commit:
Improve decoding performance

Commit:
Refactor MJPEGNode: Update output encoding to RGB8 and improve GStreamer buffer handling

Cirromulus commented on src/mjpeg_node.cpp:
```diff
@@ -0,0 +1,747 @@
+        char src_caps_str[256];
+        snprintf(src_caps_str, sizeof(src_caps_str),
+            "image/jpeg, width=%d, height=%d, pixel-aspect-ratio=1/1, framerate=%d/1",
+            1920, 1080, color_fps_);
```
can we do this with `ostream` and `operator <<` overloads? `snprintf` feels ancient :D

Cirromulus commented on src/mjpeg_node.cpp:
```diff
@@ -0,0 +1,747 @@
+                {
+                    // Don't disable GStreamer on first few frames - hardware decoder needs time to warm up
+                    // Just log and continue; the pipeline is still active and will produce frames soon
+                    RCLCPP_DEBUG(this->get_logger(), "GStreamer did not produce a frame this iteration (pipeline may be warming up)");
```
Does this warming up have an upper "broken frame" limit?

Commit:
Enhance MJPEGNode: Add compressed image publishing and increase JPEG quality

Commit:
Refactor MJPEGNode for stereo camera support and GStreamer integration

- Added support for stereo cameras with separate left and right image processing.
- Introduced new parameters for mono and stereo camera configurations, including resolution, frame rate, and JPEG quality.
- Enhanced GStreamer pipeline setup for improved decoding performance and flexibility.
- Updated ROS publishers to handle compressed images from both left and right cameras, as well as depth data.
- Refactored worker loop to accommodate new stereo processing logic and improved error handling.
- Cleaned up code for better readability and maintainability.
